### ISSUE TITLE
Opaque modifier not taken into account in the reduction algorithm

### ISSUE THREAD

**Issue by fblanqui** [Deducteam/lambdapi#547]
The following file should succeed:
```
symbol A:TYPE;
symbol a:A;
opaque symbol b ≔ a;
assertnot ⊢ a ≡ b;
```

**Comment by fblanqui** [comment]
A quick fix is to add a new field sym_opaq:bool in the type sym and, in eval, unfold a symbol definition only if sym_opaq is false.

However, in some cases, we may still want to unfold a definition. To do so, we could add an "axiom" saying that a symbol is equal to its definition, assuming the appropriate builtins on equality. Technically, this is indeed an axiom as it cannot be proved without having a symbol equivalent to its definition.

An alternative would be to change the conversion dynamically...

**Comment by fblanqui** [comment]
A simpler solution not requiring to add axioms would be, when simplifying goals, to not unfold opaque symbols, and add an unfold tactic (or have simpl takes ausmbols as arguments).

**Comment by fblanqui** [comment]
Fixed in #575 .
